lib/ansible/modules/cloud/spotinst/spotinst_aws_elastigroup.py

```diff
@@ -0,0 +1,1598 @@
+        iam_role_arn=dict(type='str'),
+        iam_role_name=dict(type='str'),
+        image_id=dict(type='str'),
+        key_pair=dict(type='str'),
+        kubernetes=dict(type='dict', required=False, default=None),
+        lifetime_period=dict(type='int'),
+        load_balancers=dict(type='list'),
+        max_size=dict(type='int'),
+        mesosphere=dict(type='dict', required=False, default=None),
+        min_size=dict(type='int'),
+        monitoring=dict(type='str'),
+        multai_load_balancers=dict(type='list'),
+        multai_token=dict(type='str'),
+        name=dict(type='str'),
+        network_interfaces=dict(type='list'),
+        on_demand_count=dict(type='int'),
+        on_demand_instance_type=dict(type='str'),
+        opsworks=dict(type='dict', required=False, default=None),
+        persistence=dict(type='dict', required=False, default=None),
+        product=dict(type='str'),
+        rancher=dict(type='dict', required=False, default=None),
+        right_scale=dict(type='dict', required=False, default=None),
+        risk=dict(type='int'),
+        roll_config=dict(type='dict', required=False, default=None),
+        scheduled_tasks=dict(type='list'),
+        security_group_ids=dict(type='list'),
+        shutdown_script=dict(type='str'),
+        signals=dict(type='list'),
+        spin_up_time=dict(type='int'),
+        spot_instance_types=dict(type='list'),
+        state=dict(default='present', choices=['present', 'absent']),
+        tags=dict(type='list'),
+        target=dict(type='int'),
+        target_group_arns=dict(type='list'),
+        tenancy=dict(type='str'),
+        terminate_at_end_of_billing_hour=dict(type='bool'),
+        unit=dict(type='str'),
+        user_data=dict(type='str'),
+        utilize_reserved_instances=dict(type='bool'),
+        uniqueness_by=dict(default='name', choices=['name', 'id']),
+        up_scaling_policies=dict(type='list'),
+        wait_for_instances=dict(required=False, type='bool', default=False),
+        wait_timeout=dict(type='int', required=False),
+    )
+
+    module = AnsibleModule(argument_spec=fields)
+
+    if not HAS_SPOTINST_SDK:
+        module.fail_json(msg="the Spotinst SDK library is required. (pip install spotinst)")
+
+    token = os.environ.get('SPOTINST_TOKEN')
+
+    if not token:
+        creds = retrieve_creds()
+        token = creds["token"]
+
+    client = spotinst.SpotinstClient(auth_token=token, print_output=False)
+
+    eg_account_id = module.params.get('account_id')
+
+    if eg_account_id is not None:
+        client = spotinst.SpotinstClient(auth_token=token, print_output=False, account_id=eg_account_id)
+
+    group_id, message, has_changed = handle_elastigroup(client=client, module=module)
+
+    instances = retrieve_group_instances(client=client, module=module, group_id=group_id)
+
+    module.exit_json(changed=has_changed, group_id=group_id, message=message, instances=instances)
+
+
+def retrieve_creds():
```

Comment:
I would move this above `def main():` to match with standard practice

Reply:
Fixed.